clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3915,10 +3925,10 @@ void Sema::CheckFunctionBodyBoundsDecls(FunctionDecl *FD, Stmt *Body) {
   }
 
   if (Cfg != nullptr) {
-    BoundsAnalysis Collector(*this, Cfg.get());
-    Collector.WidenBounds(FD);
+    BoundsAnalysis WB = Checker.getBoundsAnalysis();
+    WB.WidenBounds(FD);
```

Comment:
Can this be done within Collector.TraverseCFG?

Reply:
I guess I would have to pass FD to TraverseCFG. In any case, we would eventually get rid of this piece of code. WidenBounds would be invoked only where we actually use the widen bounds info. So I would keep this here for now.

Reply:
I thought WidenBounds would be invoked once per function declaration, either before Collector.TraverseCFG or at the beginning of Collector.TraverseCFG. I think it needs to be called before iterating over the CFG blocks and calling TraverseStmt on each block so that the bounds map is populated before CheckBoundsDeclarations needs the widened bounds information.